scripts/ci/run-integ-tests

```diff
@@ -4,17 +4,66 @@
 # binary package not from the CWD.
 
 import os
-from subprocess import check_call
+from subprocess import run, CalledProcessError
 
 _dname = os.path.dirname
 
 REPO_ROOT = _dname(_dname(_dname(os.path.abspath(__file__))))
 os.chdir(os.path.join(REPO_ROOT, 'tests'))
+NOSE_ONLY_TESTS = [
+    'test_smoke.py',
+    os.path.join('customizations', 'test_generatecliskeleton.py'),
+]
 
 
-def run(command):
-    return check_call(command, shell=True)
+class TestFailureException(Exception):
+    pass
 
 
-run('nosetests --with-xunit --cover-erase --with-coverage '
-    '--cover-package awscli --cover-xml -v integration')
+def get_nose_only_integ_tests():
+    return [
+        os.path.join('integration', integ_test)
+        for integ_test in NOSE_ONLY_TESTS
+    ]
+
+
+def run_pytest():
+    tests_to_ignore = get_nose_only_integ_tests()
+    ignore_params = [
+        f'--ignore={test_to_ignore}' for test_to_ignore in tests_to_ignore
+    ]
+    cmd = ['py.test', '-v', 'integration/'] + ignore_params
+    print(f'Running cmd: {cmd}')
+    return run(cmd)
+
+
+def run_nose():
+    test_to_run = get_nose_only_integ_tests()
+    cmd = ['nosetests', '-v'] + test_to_run
+    print(f'Running cmd: {cmd}')
+    return run(cmd)
```

Comment:
Is there any particular reason why we use subprocess instead of running tests using subprocess instead of invoke test runners directly via `pytest.main()` and `nose.run()`

Reply:
Honestly, I did not really consider it. One thing that I would be concerned about in running `main()`/`run()` directly is that any modifications from a test (e.g. a bad patch) may be carried over to next test run because it is all in the same Python process. So subprocess provides a bit more sandboxing for us. Is there a particular reason, you would want to see the test runners being invoked directly?

Reply:
The main reason is that it can give us more flexibly in test running and test error handling  during the nose -> pytest migration period.

Reply:
Yeah I don't really have too strong of a preference here. I also do not want to spend too much time focusing on specifics of this script in its current state since after this PR, I'm going to be sending another PR that ports the entire suite to pytest and simplify this script to this: https://github.com/kyleknap/aws-cli/commit/cd9bda021cec3bc49608cdbef9791436e33ba193. In the commit, I'm using `pytest.main()` and seems to be working fine. What's your thoughts on deferring making this change to the PR that includes that commit?

Reply:
Sounds good for me 👍